rdm/0006-profiles-and-flavours.md

```diff
@@ -0,0 +1,104 @@
+- Start Date: 2019-10-02
+- RFC PR: [#6](https://github.com/inveniosoftware/rfcs/pull/6)
+- Authors: Guillaume Viger
+
+# Customize created Invenio instance by choosing profile and optionally flavour
+
+## Summary
+
+When running the instance bootstrapping CLI command (`invenio-cli init`), the
+hoster should be given a choice to create an Integrated Library System (ILS) or
+Resource Data Management (RDM) "flavour" Invenio application. If RDM is chosen,
+there are further "profiles" to be chosen from, for instance "default" and
+"biomed". These choices pre-configure the application to varying levels.
+
+## Motivation
+
+> Why are we doing this? What user stories does it support? What is the expected
+outcome?
+
+As a (prospecting-)hoster, I want to have a quick working application for my
+main use case: library system or data management, so I can evaluate it
+right away (prospecting) and/or start using it (configuration/development). The
+RDM choice implies different kinds of data management for which I want to leverage
+pre-configuration / pre-customizations in order to get started faster and adopt
+same settings as other installations in my domain.
+
+As an InvenioRDM (core-, extension-, plugin-)developer for domain stakeholder,
+I want to be able to satisfy the exclusive needs/wants of my specialized
+stakeholders while still being able to develop and use domain authorities **within**
+the InvenioRDM context. I have a dedicated mechanism and space to insert
+customizations shared by instances for my domain this way.
+
+
+## Terminology
+
+**flavour**: A *flavour* represents the main use case of the application
+(e.g., ILS or RDM for now) and it translates to a different packaged application.
+The flavour packaged applications are completely different, which calls
+for a different project repository to define them.
+
+**profiles**: *Profiles* are different domains of interest (e.g., Generic or BioMed)
+for the InvenioRDM *flavour* and translate to different configurations,
+data models, and customizations. Those are not major enough to warrant being
+part of different repositories however. The profiles are defined in the same
+repository side-by-side.
+
+
+## Detailed design
+
+```console
+invenio-cli init --flavour RDM --profile biomed
```

Comment:
I was thinking that maybe the profile (or both) could be prometed to the user. Such as:

``` console
$ invenio-cli init
  - Choose your flavour?
  1) RDM
  2) ILS
  - Input choice: 1
  - You have chosen RDM flavour. Which profile do you want?
  1) BioMed
  2) Generic
```

I think is a bit more user friendly, since you prompt the user with the possibilities. Otherwise they need to do a --help to start.

Reply:
Will there be both an interactive and non-interactive `init`? We can modify this to fit what the interface the cli ends up using.

Reply:
Hmm, I haven't thought of it. Currently the cli is saving some config in a `.invenio` hidden file. However, which scenario do you foresee where filling all fields in the config instead of interactive filling them? Some sort of "automated" deployment (e.g. puppet, ansible)?

I would keep it interactive for now. We can get feedback and if people need it implement it. It wouldn't change much current code, mostly add.

Reply:
Hmm, I haven't thought of it. Currently the cli is saving some config in a .invenio hidden file. However, which scenario do you foresee where filling all fields in the config instead of interactive filling them? Some sort of "automated" deployment (e.g. puppet, ansible)?

I would keep it interactive for now. We can get feedback and if people need it implement it. It wouldn't change much current code, mostly add.

Reply:
Yup, the ansible deployment was what I had in mind or if someone knew what they were doing and wanted to skip the interactivity to just generate what they need.

I will switch to the interactive interface (something like what invenio-cli has) and we can come back to this later.

Reply:
I agree it might be needed. We can add it, and add an issue to invenio-cli. But to be done in the future?

Reply:
Yes, in the future.